src/boot/java/io/github/brassmc/brassloader/boot/discovery/ModDiscovery.java

```diff
@@ -0,0 +1,260 @@
+
+                        Path metadata = secureJar.getPath("mod.hjson");
+                        InputStream stream;
+                        if (Files.exists(metadata))
+                            stream = Files.newInputStream(metadata);
+                        else
+                            return FileVisitResult.CONTINUE;
+
+                        // Parse metadata as hjson
+                        JsonObject jsonObject = JsonValue.readHjson(new String(stream.readAllBytes())).asObject();
+
+                        // Get and validate the mandatory mod details
+                        String modid = getString(jsonObject, "modid", file);
+                        lengthCheck("modid", modid, file, 3, 20);
+                        if(!isValidModid(modid))
+                            throw new InvalidModidException("Provided modid(" + modid + ") in mod(" + file + ") must match the expression: a-z0-9/._-");
+
+                        String name = getString(jsonObject, "name", file);
+                        lengthCheck("name", name, file, 4, 32);
+
+                        String version = getString(jsonObject, "version", file);
+                        lengthCheck("version", version, file, 1, 64);
+
+                        String description = getString(jsonObject, "description", file);
+                        lengthCheck("description", description, file, 4, 2056);
+
+                        String entrypoint = getString(jsonObject, "entrypoint", file);
+
+                        String license = getString(jsonObject, "license", file);
+                        lengthCheck("license", license, file, 3, 64);
+
+                        // Get and validate the people object
+                        JsonValue peopleValue = jsonObject.get("people");
+                        if(peopleValue == null)
+                            throw new MetadataParseException("The people(including at least 1 developer) must be provided for mod: " + file);
+
+                        JsonObject peopleObj = peopleValue.asObject();
+
+                        String[] developers = getArrayOrString(peopleObj, "developers", file, true);
+                        String[] artists = getArrayOrString(peopleObj, "artists", file);
+                        String[] modellers = getArrayOrString(peopleObj, "modellers", file);
+                        String[] animators = getArrayOrString(peopleObj, "animators", file);
+                        String[] audioEngineers = getArrayOrString(peopleObj, "audioEngineers", file);
+                        String[] additionalCredits = getArrayOrString(peopleObj, "additionalCredits", file);
+
+                        // Get optional icon
+                        String iconStr = getString(jsonObject, "icon", "default.png", file);
+                        Path icon = Path.of(iconStr);
+
+                        // Get optional mixin usage (boolean)
+                        boolean usesMixins = jsonObject.getBoolean("usesMixins", false);
```

Comment:
This seems like unneeded duplication. Just check for the existence of the `mixins` block which will be implemented in the future.

Reply:
Mixins block? What do you mean? Mixins are done via the mixins config.

Reply:
I mean,
```json
"mixins": ["mymod.mixins.json"] 
```

Although, thinking about it, the `usesMixins` seems rather useless when you can just check it the mixins configs array is empty. Even then, what is the purpose of the property?

Reply:
Yea I guess on second thought, it is quite redundant.